Adamant/Services/HealthCheckWrapper.swift

```diff
@@ -150,6 +178,25 @@ private extension HealthCheckWrapper {
         healthCheck()
         lastUpdateTime = Date()
     }
+    
+    private func handleRequest<Output>(
+        requestAction: @Sendable @escaping (Service, Node) async -> Result<Output, Error>
+    ) async -> Result<Output, Error> {
+        let result = await request(requestAction)
+        return await handleResult(result: result, requestAction: requestAction)
+    }
+    
+    private func handleResult<Output>(
+        result: Result<Output, Error>,
+        requestAction: @Sendable @escaping (Service, Node) async -> Result<Output, Error>
+    ) async -> Result<Output, Error> {
+        guard case .failure(let failure) = result,
+              let apiServiceError = failure as? ApiServiceError,
```

Comment:
HealthCheckWrapper is a universal class. It shouldn't know about any particular errors. Check `HealthCheckableError`